tools/migration/sender-jid-from-mam-message.escript

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env escript
+
+-mode(compile).
+-compile(nowarn_deprecated_function).
+
+-include_lib("xmerl/include/xmerl.hrl").
+
+main([Type]) ->
+    case os:getenv("DEBUG") of
+        "1" ->
+            {ok, DebugFile} = file:open("/tmp/script-debug", [write, append]),
+            put(debug_file, DebugFile);
+        _ ->
+            ok
+    end,
+    try
+        ok = io:setopts([binary]),
+        loop(Type)
+    catch
+        C:R ->
+            debug(C, R, erlang:get_stacktrace(), #{ type => unrecoverable_error })
+    after
+        file:close(get(debug_file))
+    end;
+main(_) ->
+    print_usage().
+
+print_usage() ->
+    MyName = filename:basename(escript:script_name()),
+    io:format("Usage: ~s data-type~n", [MyName]),
+    io:format("\tdata-type = eterm | xml~n"),
+    io:format("\t\t eterm - if MIM stores messages in encoded format (default for MIM)~n"),
+    io:format("\t\t xml - if MIM stores messages in plain XML~n"),
+    io:format("\t\t Please check documentation to learn about IO format.~n~n").
+
+loop("eterm") ->
+    eterm_loop();
+loop("xml") ->
+    xml_loop();
+loop(_) -> print_usage().
+
+eterm_loop() ->
```

Comment:
You have 
`loop("eterm")` that calls `eterm_loop()` that calls `common_loop(fun eterm_loop/0, fun jid_from_eterm/1)`
and 
`loop("xml")` that calls `xml_loop()` that calls `common_loop(fun xml_loop/0, fun jid_from_xml/1)`
How about:
`loop("eterm")` that calls `common_loop(fun eterm_loop/0, fun jid_from_eterm/1)`
and `loop("xml")` that calls  `common_loop(fun xml_loop/0, fun jid_from_xml/1)`
??

Reply:
The idea is ro shorten stack trace by 1 call.